src/table/utils/__tests__/find-visible-rows.spec.ts

```diff
@@ -1,196 +1,325 @@
+        createElements(70);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findPaginationVisibleRows(rootElement, totalsPosition);
+        expect(visibleRowStartIndex).toEqual(0);
+        expect(visibleRowEndIndex).toEqual(6);
+      });
+
+      it('should return correct visible row indices when 50% of the first row is visible', () => {
+        createElements(60);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findPaginationVisibleRows(rootElement, totalsPosition);
+        expect(visibleRowStartIndex).toEqual(0);
+        expect(visibleRowEndIndex).toEqual(6);
+      });
+
+      it('should return correct visible row indices when less than 50% of the first row is visible', () => {
+        createElements(59);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findPaginationVisibleRows(rootElement, totalsPosition);
+        expect(visibleRowStartIndex).toEqual(1);
+        expect(visibleRowEndIndex).toEqual(6);
+      });
+
+      it('should return correct visible row indices when the whole last row is visible', () => {
+        createElements(-125);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findPaginationVisibleRows(rootElement, totalsPosition);
+        expect(visibleRowStartIndex).toEqual(6);
+        expect(visibleRowEndIndex).toEqual(9);
+      });
+
+      it('should return correct visible row indices when 50% of the last row is visible', () => {
+        createElements(-92.5);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findPaginationVisibleRows(rootElement, totalsPosition);
+        expect(visibleRowStartIndex).toEqual(5);
+        expect(visibleRowEndIndex).toEqual(9);
+      });
+
+      it('should return correct visible row indices when  lest than 50% of the last row is visible', () => {
+        createElements(-91.5);
+        const { visibleRowStartIndex, visibleRowEndIndex } = findPaginationVisibleRows(rootElement, totalsPosition);
+        expect(visibleRowStartIndex).toEqual(5);
+        expect(visibleRowEndIndex).toEqual(8);
+      });
     });
   });
 
-  describe('total row at top', () => {
+  describe('findVirtualizedVisibleRows', () => {
+    let rootElement: HTMLElement;
+    let tableBody: {
+      getBoundingClientRect?: () => Rect;
+      querySelectorAll?: () => any;
+    };
+    let bodyRect: Rect;
+    let cells: { getAttribute: () => string; getBoundingClientRect: () => Rect }[];
+    let viewService: ViewService;
+    const rowIndices = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+    const createCell = (rect: Rect, index: number) => {
+      return {
+        getAttribute: () => index.toString(),
+        getBoundingClientRect: () => rect,
+      };
+    };
+    const createElements = (scrollTop: number) => {
+      const rowRects: Array<Rect> = [];
+      rowIndices.forEach((item, i) => {
+        const y = i ? rowRects[i - 1].y + rowRects[i - 1].height : scrollTop;
```

Comment:
bit confusing checking that i === 0 by just checking that it is falsey. `i > 0` would be easier to read IMO